sponsorship_switzerland/views/res_partner_view.xml

```diff
@@ -10,4 +10,15 @@
             </field>
         </field>
     </record>
+
+    <record id="action_view_partner_waiting_mandate" model="ir.actions.server" >
```

Comment:
It's better to define a window action, with that you will not even need any method in recurring.contract object!
```suggestion
    <record id="action_view_partner_waiting_mandate" model="ir.actions.act_window">
        <field name="name">Waiting mandate</field>
        <field name="res_model">recurring.contract</field>
        <field name="view_type">form</field>
        <field name="view_mode">tree,form</field>
        <field name="domain" eval="[('state', '=', 'mandate'), ('mandate_date', '<', (DateTime.now() - timedelta(days=25)).strftime('%Y-%m-%d'))]"/>
    </record>
```

Reply:
that would take me a loong time to find, thank you